uui/components/overlays/__tests__/Modals.test.tsx

```diff
@@ -165,5 +166,155 @@ describe('Modals', () => {
+                    items: [
+                        {
+                            id: 'option-1',
+                            name: 'Option 1',
+                        },
+                        {
+                            id: 'option-2',
+                            name: 'Option 2',
+                        },
+                    ],
+                },
+                [],
+            );
+
+            return (
+                <ModalBlocker
+                    { ...props }
+                >
+                    <ModalWindow>
+                        <PickerInput
+                            dataSource={ dataSource }
+                            value={ option }
+                            onValueChange={ setOption }
+                            selectionMode="single"
+                            valueType="entity"
+                        />
+                    </ModalWindow>
+                </ModalBlocker>
+            );
+        }
+
+        function TestComponent(): ReactNode {
+            const handleOpenModalWindow = async (): Promise<void> => {
+                try {
+                    await testUuiCtx.uuiModals.show((
+                        modalWindowProps,
+                    ) => {
+                        return (
+                            <ModalWindowWithPickerInput
+                                { ...modalWindowProps }
+                                success={ successMock.mockImplementation(modalWindowProps.success) }
+                                abort={ abortMock.mockImplementation(modalWindowProps.abort) }
+                            />
+                        );
+                    });
+                } catch {}
+            };
+
+            return (
+                <>
+                    <Button
+                        caption="Open modal window"
+                        onClick={ handleOpenModalWindow }
+                    />
+                    <Modals />
+                </>
+            );
+        }
+
+        await renderWithContextAsync(
+            <TestComponent />,
+            {
+                wrapper,
+            },
+        );
+
+        const openModalWindowButton = await screen.findByRole(
+            'button',
+            {
+                name: /open modal window/i,
+            },
+        );
+        fireEvent.click(openModalWindowButton);
+        const pickerInput = await screen.findByRole('searchbox');
+        expect(pickerInput).toBeInTheDocument();
+
+        fireEvent.click(pickerInput);
+        // Checking that picker input's body is opened.
+        expect(
+            await screen.findByRole(
+                'option',
+                {
+                    name: /option 1/i,
+                },
+            ),
+        ).toBeVisible();
+
+        // First 'Escape' key press should close the picker input's options.
+        fireEvent.keyDown(
+            pickerInput,
+            {
+                key: 'Escape',
+            },
+        );
```

Comment:
In other tests above, I can see that a new `KeyboardEvent` is created to fire an "Escape" key press event, though `fireEvent.keyDown` works fine.